roxie/ccd/ccdserver.cpp

```diff
@@ -19358,10 +19369,10 @@ class CRoxieServerWhenActivity : public CRoxieServerActivity
         }
         else if (state != STATEstopped)
         {
-            stopDependencies(savedExtractSize, savedExtract, aborting ? WhenSuccessId : WhenFailureId);  // These ones don't get executed
-            executeDependencies(savedExtractSize, savedExtract, aborting ? WhenFailureId : WhenSuccessId); // These ones do
+            stopDependencies(savedExtractSize, savedExtract, aborted ? WhenSuccessId : WhenFailureId);  // These ones don't get executed
+            executeDependencies(savedExtractSize, savedExtract, aborted ? WhenFailureId : WhenSuccessId); // These ones do
```

Comment:
I'm not sure this is right.   This activity is not a sink, so I don't think aborted will be called.  This should probably catch exception inside nextOnGroup() and use that to set a flag if it failed.